yt_dlp/extractor/asobistage.py

```diff
@@ -0,0 +1,145 @@
+            self.raise_login_required()
+        else:
+            raise ExtractorError(f'Unknown error: {error!r}')
+
+    def _get_owned_tickets(self, video_id):
+        for url, name in [
+            ('https://asobistage-api.asobistore.jp/api/v1/purchase_history/list', 'ticket purchase history'),
+            ('https://asobistage-api.asobistore.jp/api/v1/serialcode/list', 'redemption history'),
+        ]:
+            yield from traverse_obj(self._download_json(
+                url, video_id, note=f'Downloading {name}', errnote=f'Unable to download {name}'),
+                ('payload', 'value', ..., 'digital_product_id', {str}))
+
+    def _real_initialize(self):
+        self._TOKEN = self._download_json(
+            'https://asobistage-api.asobistore.jp/api/v1/vspf/token', None,
+            note='Getting token', errnote='Unable to get token')
+
+    def _real_extract(self, url):
+        video_id, video_type_name = self._match_valid_url(url).group('id', 'type')
+        self._check_login(video_id)
+
+        webpage = self._download_webpage(url, video_id)
+
+        video_type_id = {
+            'archive': 'archives',
+            'player': 'broadcasts',
+        }.get(video_type_name)
+        if not video_type_id:
+            raise ExtractorError('Unknown video type')
+
+        event_data = self._search_nextjs_data(webpage, video_id)
+        event_id = traverse_obj(event_data, ('query', 'event', {str}))
+        event_slug = traverse_obj(event_data, ('query', 'player_slug', {str}))
+        event_title = traverse_obj(event_data, ('props', 'pageProps', 'eventCMSData', 'event_name', {str}))
+        if not all((event_id, event_slug, event_title)):
+            raise ExtractorError('Unable to get required event data')
+
+        channels_json = self._download_json(
+            f'https://asobistage.asobistore.jp/cdn/v101/events/{event_id}/{video_type_id}.json', video_id,
+            fatal=False, note='Getting channel list', errnote='Unable to get channel list')
+        available_channels = traverse_obj(channels_json, (
+            video_type_id, lambda _, v: v['broadcast_slug'] == event_slug, 'channels',
+            lambda _, v: v['chennel_vspf_id'] != '00000', {dict}))
+
+        owned_tickets = set(self._get_owned_tickets(video_id))
+        available_tickets = traverse_obj(
+            available_channels, (..., 'viewrights', ..., 'tickets', ..., 'digital_product_id', {str_or_none}))
+        if not owned_tickets.intersection(available_tickets):
+            raise ExtractorError('No valid ticket for this event', expected=True)
+
+        entries = []
+        for channel_id in traverse_obj(available_channels, (..., 'chennel_vspf_id', {str})):
+            channel_data = {}
+
+            if video_type_name == 'archive':
+                channel_json = self._download_json(
+                    f'https://survapi.channel.or.jp/proxy/v1/contents/{channel_id}/get_by_cuid', f'{video_id}/{channel_id}',
```

Comment:
why are we using `f'{video_id}/{channel_id}'` as the video_id in multiple requests but not returning it as the entry id? regardless it should be a variable
```suggestion
                    f'https://survapi.channel.or.jp/proxy/v1/contents/{channel_id}/get_by_cuid', f'{video_id}/{channel_id}',
```